### ISSUE TITLE
Version 2.0.2 of JVM packages cannot be consumed by Gradle builds

### ISSUE THREAD

**Issue by marcphilipp** [dmlc/xgboost#9787]
This change in 2.0.2 broke Gradle builds trying to resolve `ml.dmlc:xgboost4j_2.12:2.0.2`:

```
> Could not resolve ml.dmlc:xgboost4j_2.12:2.0.2.
  Required by:
      project :subproject
  > Could not resolve ml.dmlc:xgboost4j_2.12:2.0.2.
    > Could not parse POM https://repo1.maven.org/maven2/ml/dmlc/xgboost4j_2.12/2.0.2/xgboost4j_2.12-2.0.2.pom
      > Could not find ml.dmlc:xgboost-jvm_${scala.binary.version}:2.0.2.
        Searched in the following locations:
          - https://repo1.maven.org/maven2/ml/dmlc/xgboost-jvm_${scala.binary.version}/2.0.2/xgboost-jvm_${scala.binary.version}-2.0.2.pom
```

Please let me know if you require a minimal reproducer for this issue.

**Comment by wbo4958** [comment]
Yeah, it'll be great if you can share the detail to repro this issue. @marcphilipp

**Comment by wbo4958** [comment]
Wow, I can repro it. Seems something is wrong when uploading xgboost jars to the maven.

Hi @hcho3 @dotbg, Could you help look into this issue?

**Comment by trivialfis** [comment]
@wbo4958 Could you please help take a look at the nightly build and see if it works?

**Comment by wbo4958** [comment]
I used "mvn dependency:get -DgroupId=ml.dmlc -DartifactId=xgboost4j_2.12 -Dpacking=jar -Dversion=2.0.2" to try to download xgboost jars, but get the same error.

I also tried to check 2.1.0-SNAPSHOT which is ok.

The root cause is the wrong pom file for maven, please refer to https://central.sonatype.com/artifact/ml.dmlc/xgboost-jvm_2.12. 
 The code in xgboos is here https://github.com/dmlc/xgboost/blob/v2.0.2/jvm-packages/pom.xml#L8.

We can't fix this issue in 2.0.2, but we can release 2.0.3 by fixing this issue.

**Comment by trivialfis** [comment]
Let me give a shot at maven, otherwise might need some help from @hcho3 .

**Comment by pk044** [comment]
Same problem with SBT on Scala 2.13:
https://github.com/dmlc/xgboost/issues/9797#issuecomment-1827647355

**Comment by hcho3** [comment]
@wbo4958 Can I get a review for https://github.com/dmlc/xgboost/pull/9893 ?

**Comment by marcphilipp** [comment]
I can confirm that the problem is resolved with 2.0.3

**Comment by marcphilipp** [comment]
Thanks for the quick turnaround!
